src/main/java/reposense/parser/CsvParser.java

```diff
@@ -0,0 +1,135 @@
+import reposense.dataobject.RepoConfiguration;
+import reposense.frontend.CliArguments;
+import reposense.util.Constants;
+
+/**
+ * Parses a CSV configuration file for repository information.
+ */
+public class CsvParser {
+
+    private static final String MESSAGE_MALFORMED_CSV_FILE = "The supplied CSV file is malformed or corrupted.";
+    private static final String MESSAGE_UNABLE_TO_READ_CSV_FILE = "Unable to read the supplied CSV file.";
+
+    private static final int SKIP_FIRST_LINE = 1;
+
+    /** Positions of the elements of a line in the user-supplied CSV file */
+    private static final int ORGANIZATION_POSITION = 0;
+    private static final int REPOSITORY_NAME_POSITION = 1;
+    private static final int BRANCH_POSITION = 2;
+    private static final int GITHUB_ID_POSITION = 3;
+    private static final int DISPLAY_NAME_POSITION = 4;
+    private static final int ALIAS_POSITION = 5;
+
+    /** String format of the key for RepoConfig HashMap */
+    private static final String REPO_CONFIG_MAP_KEY_FORMAT = "%s|%s|%s";
+
+    private HashMap<String, RepoConfiguration> repoMap = new HashMap<String, RepoConfiguration>();
+
+    /**
+     * Creates a RepoConfiguration object in repoMap, if it does not exists.
+     *
+     * Parameters - organization, repositoryName and branch are used to create the key to access RepoMap.
+     */
+    private void createRepoConfigInMapIfNotExists(final String organization, final String repositoryName,
+            final String branch, final Date sinceDate, final Date untilDate) {
+
+        final String key = String.format(REPO_CONFIG_MAP_KEY_FORMAT, organization, repositoryName, branch);
+
+        if (!repoMap.containsKey(key)) {
+            RepoConfiguration config = new RepoConfiguration(organization, repositoryName, branch);
+
+            config.setSinceDate(sinceDate);
+            config.setUntilDate(untilDate);
+
+            repoMap.put(key, config);
+        }
+    }
+
+    /**
+     * Creates a RepoConfiguration object in the repoMap, if it does not exists.
+     *
+     * Parameters - organization, repositoryName and branch are used to create the key to access RepoMap.
+     */
+    private RepoConfiguration getRepoConfigFromMap(final String organization, final String repositoryName,
+            final String branch, final Date sinceDate, final Date untilDate) {
+
+        final String key = String.format(REPO_CONFIG_MAP_KEY_FORMAT, organization, repositoryName, branch);
+        createRepoConfigInMapIfNotExists(organization, repositoryName, branch, sinceDate, untilDate);
+
+        return repoMap.get(key);
+    }
+
+    /**
+     * Returns a list of RepoConfiguration, which are the inflated object a line of the csv file.
+     *
+     * @throws IllegalArgumentException If parameter is null or user-supplied csv file fails to parse.
+     */
+    public List<RepoConfiguration> parse(CliArguments argument) throws IllegalArgumentException {
+
+        assert (argument != null);
```

Comment:
`@nonnull` is probably more appropriate here.

Reply:
Not using @NotNull for now. Keep in view.